### DISCUSSION TITLE
Show build results also for forks

### DISCUSSION THREAD

**Discussion by msirringhaus** [openSUSE/openSUSE-git#40] [Ideas]
With the traditional OBS-workflow, I branch a package in OBS, do updates, send it back to my branch and see if it builds fine on all archs and relevant code-streams. Sometimes I try out the resulting rpms. And only once I'm satisfied, do I issue an SR.

Currently, the git-integration seems to only actually start building something, once I've created the final SR/PR to Factory.

It would be good to get builds for forks, too.

Possible options:
1. Just send an email with a link to the OBS-buildenv for each commit
2. Have some kind of CI button that allows me to view it directly from src.opensuse.org

**Comment by dcermak** [comment]
Would you be fine with having an osc plugin for this? I think creating builds automatically on commits can get very expensive soon.

**Reply by mcepl** [reply]
For me, absolutely. Actually, would `osc rebuild` work? Also, talking about `osc rebuild` generally, it would be lovely to have `--force` option, so that I wouldn’t have to fun `osc wipebinaries --all ; and osc rebuild --all` all the time.

**Reply by msirringhaus** [reply]
So, something like `osc buildfork` (or `osc rebuild` or whatnot) to trigger a build on demand? And then it returns a URL to the build-env? Or hook into `osc r` et al.? Or both?

**Comment by AdamMajer** [comment]
I think this is doable, kind of. One problem that arises is when you have non-packaging software on src.opensuse.org, so you can't just assume things. :wink:

FWIW, OBS currently builds on pushes, which is what "osc commit" actually is. This makes it doable. In either case, it gets messy quickly when we have projects in Git. More thinking needed here because OBS hides a lot of complexity and exposing this complexity is a double edged sword.

**Comment by mcepl** [comment]
On Thu Jun 27, 2024 at 11:24 AM CEST, msirringhaus wrote:
> And then it return a URL to the build-env?

Not sure what you mean.

**Reply by msirringhaus** [reply]
Presumably, this plugin would create some new repo somewhere on OBS instead of using my home-branch to build in (similar to what happens already when doing a PR to the factory-branch: `devel:Factory:git-workflow:staging:...`). I would need to know where I can find that, hence: Returning an URL to the location on OBS where it will be built.

**Comment by hennevogel** [comment]
If we invent a "fork" [filter](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration#sec.obs.obs_scm_ci_workflow_integration.setup.obs_workflows.filters) all of this is solved with SCM/CI Workflows. The Gitea organization could setup any workflow that they want to happen for pushes to forks.

**Reply by hennevogel** [reply]
BTW like basically all of these use cases are solved with SCM/CI Workflows already. Including the branching, code syncing, creating submit requests  etc. etc. No idea why this is ignored so hard... 😸

**Reply by dcermak** [reply]
We do not necessarily want builds to happen automatically on every single fork, but setting this up was something that I wanted to use.

**Reply by hennevogel** [reply]
Why not? We do the same today, you branch a package you get a build.

**Comment by mcepl** [comment]
On Fri Jun 28, 2024 at 12:45 PM CEST, msirringhaus wrote:
> Presumably, this plugin would create some new repo somewhere
> on OBS instead of using my home-branch to build in (similar to
> what happens already when doing a PR to the factory-branch:
> `devel:Factory:git-workflow:staging:...`). I would need to know
> where I can find that, hence: Returning an URL to the location on
> OBS where it will be built.

Oh, you are talking about the initial creation of your
fork. That's different issue, we are talking here about the
equivalent of `osc build`.

**Reply by msirringhaus** [reply]
No, we are not ;-)
From my initial message:
> send it back to my branch and see if it builds fine on all archs and relevant code-streams

I'm talking about "building my fork on OBS", not about "building locally" (which, yes, would also be nice).
